### DISCUSSION TITLE
Enforcing conventional commits in konveyor

### DISCUSSION THREAD

**Discussion by djzager** [konveyor/enhancements#89] [Polls]
We are beginning an effort in [Konveyor CI and Release Engineering](https://github.com/orgs/konveyor/projects/47) to automate the release process. Starting with https://github.com/konveyor/tackle2-hub/issues/201, we will be introducing workflows to create releases for us with minimal (goal is no) human interaction.

One of the important components of that issue is the ability to automatically generate meaningful changelogs based on the commit history of the project and [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) are an excellent way for us to do this.

**How does this impact me?**

If we choose conventional commits, we will eventually expect all contributions to projects in the Konveyor community to abide by that convention.

**Poll Question**

Restating the Poll here where I still have emoji support.

Which variation of conventional commits should we use in konveyor?

1. Emojis please! For example `fix` -> :bug:, `feat` -> :sparkles: , `feat!` (breaking change) -> :warning: 
2. Emojis are weird. Just follow the conventions.
3. Please don't. If you are strongly against conventional commits for some reason, say so (and please include why and/or alternatives).

**Comment by savitharaghunathan** [comment]
✨

**Comment by jortel** [comment]
I'm not sure I understand the Emojis part of the poll.  I did not see Emojis/styles mentioned in the conventional commit document. Can someone elaborate?

**Reply by djzager** [reply]
The emojis themselves are not part of the document or convention. The idea would be agree on the emoji to communicate the same idea, like:

| Type     | Emoji                 | code                    |
|:---------|:----------------------|:------------------------|
| feat     | :sparkles:            | `:sparkles:`            |
| fix      | :bug:                 | `:bug:`                 |
| docs     | :books:               | `:books:`               |
| style    | :gem:                 | `:gem:`                 |
| refactor | :hammer:              | `:hammer:`              |
| perf     | :rocket:              | `:rocket:`              |
| test     | :rotating_light:      | `:rotating_light:`      |
| build    | :package:             | `:package:`             |
| ci       | :construction_worker: | `:construction_worker:` |
| chore    | :wrench:              | `:wrench:`              |

It isn't better, it is just that some have stated they like the emojis.

**Reply by mansam** [reply]
To be clear, is the intention that these emoji would be in the commit message itself instead of the text label for the commit type?

**Reply by djzager** [reply]
In the commit message itself.

**Reply by jortel** [reply]
So the choice is between:
```
feat: allow provided config object to extend other configs.
```
and
```
:sparkles: allow provided config object to extend other configs
```
Or something else?

**Reply by djzager** [reply]
Yes. Those are the choices and it wasn't until now that I realized I was missing a full example. My fault. Sorry about that.

**Reply by mansam** [reply]
Understood, thanks. I'm opposed to the emojis then. They don't show up well on my display and the meanings aren't intuitive to me.

**Reply by ibolton336** [reply]
I don't have a strong preference either way. When working on Patternfly, they used a similar approach for labeling PRs  https://github.com/patternfly/patternfly-react/blob/main/CONTRIBUTING.md#pr-labels . I am FOR a more organized approach for labeling PRs - thanks for getting this together @djzager

**Comment by mturley** [comment]
@djzager after some time using them, I've decided I'm not a fan of the emojis.
* When used directly in a commit message, opening a PR defaults the title to using the actual unicode character and not the `:sparkles:` text, which fails CI.
* The meanings are not intuitive, I have to look them up often.

**Reply by shawn-hurley** [reply]
@mturley if you use `:sparkles:` in the commit I think it works at least from the command line. Don't know if you are using an IDE or something that is pre-rendering it?
Example: https://github.com/konveyor/analyzer-lsp/pull/131/commits/5667ea2655d08dfad69c8f0c9c8aa80026a115b4

**Reply by mturley** [reply]
I'm using the CLI, and the commit message appears correctly in git log. I am using the `gh` CLI to create the PR, maybe that's the culprit.

**Reply by shawn-hurley** [reply]
Yeah, I don't know. I just wanted to add my experience.

For the discoverability problem, though, I don't have a good answer so, maybe we have to brainstorm that or consider switching

**Reply by djzager** [reply]
Going to see how well [this](https://github.com/konveyor/tackle2-ui/pull/850) works, if at all, and see where we go from there.

**Reply by mturley** [reply]
The issue I was seeing seems to have disappeared 🙃 and the PR template does help with discoverability. I'm okay with keeping it.

**Reply by shawn-hurley** [reply]
Love the idea of discoverability being in the template +100 @djzager
